main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -1541,8 +1537,7 @@ object Desugar {
     val e = visitExp(exp0)
     val prefix = mkDebugPrefix(e, kind0, loc0)
     val e1 = DesugaredAst.Expr.Cst(Constant.Str(prefix), loc0)
-    val call = mkApplyFqn("Debug.debugWithPrefix", List(e1, e), loc0)
-    DesugaredAst.Expr.UncheckedMaskingCast(call, loc0)
+    mkApplyFqn("Debug.debugWithPrefix", List(e1, e), loc0)
```

Comment:
Does this require a cast or no?

Reply:
I've changed debugWithPrefix to be pure and have a cast inside .. but actually maybe that's a bad idea

Reply:
Thats seems worse to me.